overlord/snapstate/snapstate.go

```diff
@@ -1297,6 +1297,15 @@ func canRemove(si *snap.Info, snapst *SnapState, removeAll bool) bool {
 		return true
 	}
 
+	// Allow snap.TypeOS removals if a different base is in use
+	if si.Type == snap.TypeOS {
+		if model, err := Model(st); err == nil {
+			if model.Base() != "" {
```

Comment:
What if model.Base == si.Name? Is that captured by other code or is this check too simple?

Reply:
Thats a nice idea. My initial thinking was to ensure this via https://github.com/snapcore/snapd/pull/5358 but when reading this code people will probably wonder.

Reply:
I added a comment now that explains about boot-bases.